server/proxy/install.go

```diff
@@ -189,34 +187,21 @@ func (p *Proxy) ensureBot(app *apps.App, actingUserID string, client *model.Clie
 		return errors.Wrap(err, "failed set bot icon")
 	}
 
-	if app.RequestedPermissions.Contains(apps.PermissionActAsBot) {
-		var token *model.UserAccessToken
-		if app.BotAccessTokenID != "" {
-			token, response = client.GetUserAccessToken(app.BotAccessTokenID)
-			if response.Error != nil {
-				return errors.Wrap(response.Error, "failed to create bot user's access token")
-			}
-			if response.StatusCode != http.StatusOK {
-				return fmt.Errorf("failed to create bot user's access token, status code = %v", response.StatusCode)
-			}
-		} else {
-			token, response = client.CreateUserAccessToken(fullBot.UserId, "Mattermost App Token")
-			if response.Error != nil {
-				return errors.Wrap(response.Error, "failed to create bot user's access token")
-			}
-			if response.StatusCode != http.StatusOK {
-				return fmt.Errorf("failed to create bot user's access token, status code = %v", response.StatusCode)
-			}
+	// Create an access token on a fresh app install
+	if app.RequestedPermissions.Contains(apps.PermissionActAsBot) &&
+		app.BotAccessTokenID == "" {
+		token, response := client.CreateUserAccessToken(fullBot.UserId, "Mattermost App Token")
+		if response.Error != nil {
+			return errors.Wrap(response.Error, "failed to create bot user's access token")
+		}
+		if response.StatusCode != http.StatusOK {
+			return fmt.Errorf("failed to create bot user's access token, status code = %v", response.StatusCode)
 		}
```

Comment:
This part fixes https://mattermost.atlassian.net/browse/MM-36335.

`GetUserAccessToken` doesn't return `token.Token` for security reasons and hence `app.BotAccessToken` will be empty. Given that there is no need to fetch the token again (it's in the store)  the code only creates one on a fresh install.